src/Normalizer/MoocListNormalizer.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace AppBundle\Normalizer;
+
+use AppBundle\Entity\Mooc\Mooc;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+class MoocListNormalizer implements NormalizerInterface
+{
+    private const FORMAT = 'json';
+
+    /** @var RouterInterface */
+    private $router;
+
+    public function __construct(RouterInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    public function normalize($object, $format = null, array $context = [])
+    {
+        return $this->normalizeMooc($object);
+    }
+
+    public function supportsNormalization($data, $format = null)
+    {
+        return self::FORMAT === $format && $data instanceof Mooc;
+    }
+
+    private function normalizeMooc(Mooc $mooc): array
+    {
+        return [
+            'chapter_count' => $mooc->countChapters(),
```

Comment:
I think you don't need to manage other fields, but just `image`. You can decorate ItemNormalizer of ApiPlatform to do this.